### PR TITLE
One instance

### PR DESCRIPTION
**Pull request by pemba1s1**
### Issue
- #143 

### Changes
- Run only one instance of freescribe
- Only one instance of DebugWindow

## Summary by Sourcery

Restrict the FreeScribe application and DebugWindow to a single instance by implementing checks for existing instances and bringing them to the front if they are already running.

Enhancements:
- Ensure only one instance of the FreeScribe application can run at a time by checking for existing instances and bringing them to the front if found.
- Modify the DebugWindow to prevent multiple instances from opening and bring the existing window to the front if it is already open.

### CODE REVIEW

File: src/FreeScribe.client/utils/utils.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
+
+from ctypes import WinDLL
```
[the remark this answers is not in the document]
Comment (pemba1s1) [reply src/FreeScribe.client/utils/utils.py:2]: to be done in the future (when making changes for mac)

File: src/FreeScribe.client/utils/utils.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    """
+    Check if another instance of the application is already running.
+    Parameters:
+        app_name (str): The name of the application
+    Returns:
+        bool: True if another instance is running, False otherwise
+    """
+    U32DLL = WinDLL('user32')
+    # get the handle of any window matching 'app_name'
+    hwnd = U32DLL.FindWindowW(None, app_name)
```
[the remark this answers is not in the document]
Comment (pemba1s1) [reply src/FreeScribe.client/utils/utils.py:28]: nice suggestions. implemented it

**Comment by pemba1s1** [comment]
@sourcery-ai review

File: src/FreeScribe.client/utils/utils.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    # Define the mutex name
+    MUTEX_NAME = 'Global\\FreeScribe_Instance'
+    ERROR_ALREADY_EXISTS = 183
+
+    # Create a named mutex
+    mutex = ctypes.windll.kernel32.CreateMutexW(None, False, MUTEX_NAME)
+    already_running = ctypes.windll.kernel32.GetLastError() == ERROR_ALREADY_EXISTS
+    return already_running
+
+def bring_to_front(app_name: str):
```
[the remark this answers is not in the document]
Comment (pemba1s1) [reply src/FreeScribe.client/utils/utils.py:19]: to be done in future when working on mac version.

File: src/FreeScribe.client/utils/utils.py [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    Check if another instance of the application is already running.
+    Returns:
+        bool: True if another instance is running, False otherwise
+    """
+    # Define the mutex name
+    MUTEX_NAME = 'Global\\FreeScribe_Instance'
+    ERROR_ALREADY_EXISTS = 183
+
+    # Create a named mutex
+    mutex = ctypes.windll.kernel32.CreateMutexW(None, False, MUTEX_NAME)
```
[the remark this answers is not in the document]
Comment (pemba1s1) [reply src/FreeScribe.client/utils/utils.py:21]: mutex is closed on application quit

**Review by yingbull — APPROVED** [review]

**Pull request merged**